File: solution/3400-3499/3467.Transform Array by Parity/Solution.go
func transformArray(nums []int) []int {
	even := 0
	for _, x := range nums {
		even += x&1 ^ 1
	}
	for i := 0; i < even; i++ {
		nums[i] = 0
	}
	for i := even; i < len(nums); i++ {
		nums[i] = 1
	}
	return nums
}

File: solution/3400-3499/3467.Transform Array by Parity/Solution.ts
function transformArray(nums: number[]): number[] {
    const even = nums.filter(x => x % 2 === 0).length;
    for (let i = 0; i < even; ++i) {
        nums[i] = 0;
    }
    for (let i = even; i < nums.length; ++i) {
        nums[i] = 1;
    }
    return nums;
}
